### PR TITLE
ref(pii): Expose source example data to frontend

### PR DESCRIPTION
**Pull request by untitaker**
We want to help the user understand what they're actually going to scrub by showing example data of PII sources/sourcenames in the frontend. This is a follow-up to https://github.com/getsentry/relay/pull/575

### CODE REVIEW

File: src/sentry/static/sentry/app/views/settings/components/dataPrivacyRules/dataPrivacyRulesForm/dataPrivacyRulesFormSource.tsx [hunk 1]
```diff
@@ -373,23 +358,37 @@ class DataPrivacyRulesFormSource extends React.Component<Props, State> {
...
-            {suggestions.slice(0, 50).map((suggestion, index) => (
-              <SuggestionItem
-                key={suggestion.value}
-                onClick={this.handleClickSuggestionItem(suggestion)}
-                active={index === activeSuggestion}
-                tabIndex={-1}
-              >
-                <TextOverflow>{suggestion.value}</TextOverflow>
-                {suggestion?.description && (
+          <div>
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/views/settings/components/dataPrivacyRules/dataPrivacyRulesForm/dataPrivacyRulesFormSource.tsx:361]:
```suggestion
          <React.Fragment>
```

File: src/sentry/static/sentry/app/views/settings/components/dataPrivacyRules/dataPrivacyRulesForm/sourceSuggestionExamples.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+class SourceSuggestionExamples extends React.Component<Props, State> {
+  state: State = {
+    isOpen: false
+  };
+
+  toggleModal = () => {
+    this.setState({isOpen: !this.state.isOpen});
+  };
+
+  stopPropagation = (e) => {
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/views/settings/components/dataPrivacyRules/dataPrivacyRulesForm/sourceSuggestionExamples.tsx:27]:
```suggestion
  stopPropagation = (e: React.MouseEvent<HTMLSpanElement>) => {
```

File: src/sentry/static/sentry/app/views/settings/components/dataPrivacyRules/dataPrivacyRulesForm/sourceSuggestionExamples.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    // Necessary to stop propagation of click events from modal that we can't
+    // catch otherwise.
+    e.stopPropagation();
+  }
+
+  render() {
+    const {isOpen} = this.state;
+    const {examples, sourceName} = this.props;
+
+    if (examples.length == 0) {
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/views/settings/components/dataPrivacyRules/dataPrivacyRulesForm/sourceSuggestionExamples.tsx:37]:
```suggestion
    if (examples.length === 0) {
```

File: src/sentry/static/sentry/app/views/settings/components/dataPrivacyRules/dataPrivacyRulesForm/sourceSuggestionExamples.tsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+      return (<span></span>);
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/views/settings/components/dataPrivacyRules/dataPrivacyRulesForm/sourceSuggestionExamples.tsx:38]: I would return null here, but better yet... if examples.length === 0 this component should not be rendered

File: src/sentry/static/sentry/app/views/settings/components/dataPrivacyRules/dataPrivacyRulesForm/types.tsx [hunk 5]
```diff
@@ -40,4 +40,5 @@ export type SourceSuggestion = {
   type: SourceSuggestionType;
   value: string;
   description?: string;
+  examples?: Array<any>;
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/views/settings/components/dataPrivacyRules/dataPrivacyRulesForm/types.tsx:43]:
```suggestion
  examples?: Array<string>;
```
?

File: src/sentry/static/sentry/app/views/settings/components/dataPrivacyRules/dataPrivacyRulesForm/dataPrivacyRulesFormSource.tsx [hunk 6]
```diff
@@ -373,23 +358,37 @@ class DataPrivacyRulesFormSource extends React.Component<Props, State> {
...
+                    <SourceSuggestionExamples
+                      examples={suggestion.examples}
+                      sourceName={suggestion.value}
+                    />
+                  )}
+                </SuggestionItem>
+              ))}
+            </SuggestionsWrapper>
+            <SuggestionsOverlay onClick={this.handleClickOutside} />
+          </div>
```
Comment (untitaker) [inline src/sentry/static/sentry/app/views/settings/components/dataPrivacyRules/dataPrivacyRulesForm/dataPrivacyRulesFormSource.tsx:391]:
```suggestion
          </React.Fragment>
```

File: src/sentry/static/sentry/app/views/settings/components/dataPrivacyRules/dataPrivacyRulesForm/sourceSuggestionExamples.tsx [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+    z-index: 1003;
+    @media (max-width: ${p => p.theme.breakpoints[0]}) {
+      width: 100%;
+    }
+  }
+
+  .modal-body {
+    max-height: 500px;
+    overflow-y: auto;
+    padding: 20px 30px;
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/views/settings/components/dataPrivacyRules/dataPrivacyRulesForm/sourceSuggestionExamples.tsx:76]:
```suggestion
    padding: ${space(3)} ${space(4)};
```

File: src/sentry/static/sentry/app/views/settings/components/dataPrivacyRules/dataPrivacyRulesForm/sourceSuggestionExamples.tsx [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+    margin: -20px -30px;
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/views/settings/components/dataPrivacyRules/dataPrivacyRulesForm/sourceSuggestionExamples.tsx:77]:
```suggestion
    margin: -${space(3)} -${space(4)};
```

**Review by priscilawebdev — COMMENTED** [review]
Great Job!! Almost there 😉

**Review by priscilawebdev — APPROVED** [review]
LGTM 👏

**Pull request merged**